tensorflow/compiler/tests/random_ops_test.py

```diff
@@ -116,12 +116,22 @@ def testTruncatedNormalIsNotConstant(self):
     def rng(dtype):
       return random_ops.truncated_normal(shape=[2], dtype=dtype)
 
-    self._testRngIsNotConstant(rng, dtypes.float32)
+    # There is no double precision float kernel for truncated normal
+    for dtype in self._random_types() & self.float_types - {np.float64}:
+      self._testRngIsNotConstant(rng, dtype)
 
   def testTruncatedNormalIsInRange(self):
-    count = 10000000
-    # TODO(b/34339814): make this test work with 16 bit float types.
-    for dtype in self._random_types() & {dtypes.float32, dtypes.float64}:
+    count = 1000000
+
+    # There is no double precision float kernel for truncated normal
+    for dtype in self._random_types() & self.float_types - {np.float64}:
+
+      # TODO: CPU backend doesn't produce a valid distribution. Remove this
+      # check when it does.
+      if (self.device in ["XLA_GPU", "XLA_CPU"
```

Comment:
I added this because the CPU doesn't give completely valid values for half.  i'm guessing that this is the same issue as is referenced on line 103

Reply:
note though that the CPU does pass the 'truncated normal random values are different' test which has been enabled.